components/docs/DocsCards.tsx

```diff
@@ -0,0 +1,73 @@
+import Link from 'next/link';
+import React from 'react';
+
+import { HeadingLevel, HeadingTypeStyle } from '@/types/typography/Heading';
+import { ParagraphTypeStyle } from '@/types/typography/Paragraph';
+
+import { buckets } from '../data/buckets';
+import Heading from '../typography/Heading';
+import Paragraph from '../typography/Paragraph';
+
+type IconType = (props: { className: string }) => React.ReactElement;
+
+interface CardProps {
+  title: string;
+  description: string;
+  link: string;
+  className: string;
+  Icon: IconType;
+}
+
+/**
+ * @description This component displays a card with title, description, link, and an associated icon.
+ *
+ * @component
+ * @param {CardProps} props - The props for the Card component.
+ * @param {string} props.title - The title of the card.
+ * @param {string} props.description - The description of the card.
+ * @param {string} props.link - The link associated with the card.
+ * @param {string} props.className - CSS class for styling the card.
+ * @param {IconType} props.Icon - The Icon component for the card.
+ */
+function Card({ title, description, link, className, Icon } : CardProps): JSX.Element {
```

Comment:
@akshatnema We should not use `React.ReactNode` here because it's being used inside the JSX in `DocsCards`. Due to this, using `React.ReactNode` is causing a build error.

Reply:
Then create a separate component as `DocCard.tsx`.

Reply:
The functional components shouldn't be created in another functional component.

Reply:
Done